src/store.c

```diff
@@ -239,8 +276,11 @@ bool rill_store_rm(struct rill_store *store)
 static bool writer_open(
         struct rill_store *store,
         const char *file,
-        struct vals *vals, size_t pairs,
-        rill_ts_t ts, size_t quant)
+        struct vals *vals,
+        struct vals *inverted_vals,
```

Comment:
I would use names that match the column a/b nomenclature to make this a bit less confusing. So something like `values_a`, `values_b` or `struct vals **values` where `values[rill_col_a]` is used to retrieve the right column.